stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -270,6 +270,29 @@ lazy_static! {
     ];
 }
 
+// Cause an artifical stall in block-processing, for testing.
+#[cfg(any(test, feature = "testing"))]
```

Comment:
Can you make this just `cfg(test)`? We don't use this stall flag outside of this crate (for now), so `cfg(test)` is simpler.

Also, this doesn't build (outside of tests) as it currently stands, something like this maybe would be simpler:

```
diff --git a/stackslib/src/chainstate/nakamoto/coordinator/tests.rs b/stackslib/src/chainstate/nakamoto/coordinator/tests.rs
index ae6ce9911..569114aa1 100644
--- a/stackslib/src/chainstate/nakamoto/coordinator/tests.rs
+++ b/stackslib/src/chainstate/nakamoto/coordinator/tests.rs
@@ -44,12 +44,11 @@ use crate::chainstate::nakamoto::coordinator::load_nakamoto_reward_set;
 use crate::chainstate::nakamoto::miner::NakamotoBlockBuilder;
 use crate::chainstate::nakamoto::signer_set::NakamotoSigners;
 use crate::chainstate::nakamoto::test_signers::TestSigners;
+use crate::chainstate::nakamoto::test_stall::*;
 use crate::chainstate::nakamoto::tests::get_account;
 use crate::chainstate::nakamoto::tests::node::TestStacker;
 use crate::chainstate::nakamoto::{
-    disable_process_block_stall, enable_process_block_stall, NakamotoBlock,
-    NakamotoBlockObtainMethod, NakamotoChainState, NakamotoStagingBlocksConnRef,
-    TEST_PROCESS_BLOCK_STALL,
+    NakamotoBlock, NakamotoBlockObtainMethod, NakamotoChainState, NakamotoStagingBlocksConnRef,
 };
 use crate::chainstate::stacks::address::PoxAddress;
 use crate::chainstate::stacks::boot::pox_4_tests::{get_stacking_minimum, get_tip};
diff --git a/stackslib/src/chainstate/nakamoto/mod.rs b/stackslib/src/chainstate/nakamoto/mod.rs
index b2106d8e7..b1a59729d 100644
--- a/stackslib/src/chainstate/nakamoto/mod.rs
+++ b/stackslib/src/chainstate/nakamoto/mod.rs
@@ -272,25 +272,27 @@ lazy_static! {
 
 // Cause an artifical stall in block-processing, for testing.
 #[cfg(any(test, feature = "testing"))]
-pub static TEST_PROCESS_BLOCK_STALL: std::sync::Mutex<Option<bool>> = std::sync::Mutex::new(None);
-
-fn stall_block_processing() {
-    if *TEST_PROCESS_BLOCK_STALL.lock().unwrap() == Some(true) {
-        // Do an extra check just so we don't log EVERY time.
-        warn!("Block processing is stalled due to testing directive.");
-        while *TEST_PROCESS_BLOCK_STALL.lock().unwrap() == Some(true) {
-            std::thread::sleep(std::time::Duration::from_millis(10));
+mod test_stall {
+    pub static TEST_PROCESS_BLOCK_STALL: std::sync::Mutex<Option<bool>> = std::sync::Mutex::new(None);
+
+    pub fn stall_block_processing() {
+        if *TEST_PROCESS_BLOCK_STALL.lock().unwrap() == Some(true) {
+            // Do an extra check just so we don't log EVERY time.
+            warn!("Block processing is stalled due to testing directive.");
+            while *TEST_PROCESS_BLOCK_STALL.lock().unwrap() == Some(true) {
+                std::thread::sleep(std::time::Duration::from_millis(10));
+            }
+            info!("Block processing is no longer stalled due to testing directive.");
         }
-        info!("Block processing is no longer stalled due to testing directive.");
     }
-}
 
-pub fn enable_process_block_stall() {
-    TEST_PROCESS_BLOCK_STALL.lock().unwrap().replace(true);
-}
+    pub fn enable_process_block_stall() {
+        TEST_PROCESS_BLOCK_STALL.lock().unwrap().replace(true);
+    }
 
-pub fn disable_process_block_stall() {
-    TEST_PROCESS_BLOCK_STALL.lock().unwrap().replace(false);
+    pub fn disable_process_block_stall() {
+        TEST_PROCESS_BLOCK_STALL.lock().unwrap().replace(false);
+    }
 }
 
 /// Trait for common MARF getters between StacksDBConn and StacksDBTx
@@ -1746,7 +1748,7 @@ impl NakamotoChainState {
         dispatcher_opt: Option<&'a T>,
     ) -> Result<Option<StacksEpochReceipt>, ChainstateError> {
         #[cfg(any(test, feature = "testing"))]
-        stall_block_processing();
+        test_stall::stall_block_processing();
 
         let nakamoto_blocks_db = stacks_chain_state.nakamoto_blocks_db();
         let Some((next_ready_block, block_size)) =
```

Reply:
Oops. Fixed in 571dfee691e1eb8a6f5978331296156104d62c4a.